src/mono/wasm/runtime/loader/blazor/_Integration.ts

```diff
@@ -199,7 +208,7 @@ export function mapBootConfigToMonoConfig(moduleConfig: MonoConfigInternal, appl
         if (config === "appsettings.json" || config === `appsettings.${applicationEnvironment}.json`) {
             assets.push({
                 name: config,
-                resolvedUrl: appendUniqueQuery((document ? document.baseURI : "/") + config, "vfs"),
+                resolvedUrl: appendUniqueQuery(toAbsoluteBaseUri(config), "vfs"),
```

Comment:
Could we resolve the relative path at build time and put it into the boot config so that we don't have to do base URI things ?

Reply:
We don't know baseURI during build.

Reply:
but we know the relative location ?

Reply:
Are you suggesting to generate the URL as `../appsettings.json`?

Reply:
I don't know if that's good idea. Just that base URI seems fragile to me

Reply:
Discussed offline
- Generate `../appsettings.json`
- Change `locateFile` to use `new URI(relativePath, loaderHelpers.scriptDirectory)` (if URI is available)